lib/src/Status/QClipboardProxy.cpp

```diff
@@ -1,14 +1,69 @@
 #include "DOtherSide/Status/QClipboardProxy.h"
 
-#include <QGuiApplication>
+#include <QBuffer>
 #include <QClipboard>
+#include <QGuiApplication>
+#include <QImage>
+#include <QMimeData>
+#include <QUrl>
 
 QClipboardProxy::QClipboardProxy()
+    : m_clipboard(QGuiApplication::clipboard())
 {
-    connect(QGuiApplication::clipboard(), &QClipboard::dataChanged, this, &QClipboardProxy::textChanged);
+    connect(m_clipboard, &QClipboard::changed, this, [this](QClipboard::Mode mode) {
+        if (mode == QClipboard::Clipboard)
+            emit contentChanged();
+    });
+}
+
+bool QClipboardProxy::hasText() const
+{
+    return m_clipboard->mimeData()->hasText();
 }
 
 QString QClipboardProxy::text() const
 {
-    return QGuiApplication::clipboard()->text();
+    return m_clipboard->text();
+}
+
+bool QClipboardProxy::hasHtml() const
+{
+    return m_clipboard->mimeData()->hasHtml();
+}
+
+QString QClipboardProxy::html() const
+{
+    return m_clipboard->mimeData()->html();
+}
+
+bool QClipboardProxy::hasImage() const
+{
+    return m_clipboard->mimeData()->hasImage();
+}
+
+QImage QClipboardProxy::image() const
+{
+    return m_clipboard->image();
+}
+
+QByteArray QClipboardProxy::imageBase64() const
+{
+    if (!hasImage())
+        return {};
+
+    const auto img = image();
+    QByteArray byteArray;
+    QBuffer buffer(&byteArray);
+    img.save(&buffer, "PNG"); // writes the image in PNG format inside the buffer
```

Comment:
Can we determine whether we really have a PNG? This could be JPG or WEBP as well

Reply:
Here we always save as PNG but we could yeah, TODO :)

Reply:
Might be problematic for GIFs... we would effectively kill the animation :/

Reply:
FWIW I don't think we support sending GIFs as image messages atm

Reply:
I stand corrected. It looks like we do